jni/src/faiss_wrapper.cpp

```diff
@@ -170,39 +184,41 @@ void knn_jni::faiss_wrapper::InsertToIndex(knn_jni::JNIUtilInterface * jniUtil,
 }
 
 void knn_jni::faiss_wrapper::WriteIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env,
-                                         jstring indexPathJ, jlong index_ptr, IndexService* indexService) {
+                                        jobject output, jlong index_ptr, IndexService* indexService) {
 
-    if (indexPathJ == nullptr) {
-        throw std::runtime_error("Index path cannot be null");
+    if (output == nullptr) [[unlikely]] {
+        throw std::runtime_error("Index output stream cannot be null");
     }
 
-    // Index path
-    std::string indexPathCpp(jniUtil->ConvertJavaStringToCppString(env, indexPathJ));
+    // IndexOutput wrapper.
+    knn_jni::stream::NativeEngineIndexOutputMediator mediator {jniUtil, env, output};
+    knn_jni::stream::FaissOpenSearchIOWriter writer {&mediator};
 
-    // Create index
-    indexService->writeIndex(indexPathCpp, index_ptr);
+    // Create index.
+    indexService->writeIndex(&writer, index_ptr);
+    mediator.flush();
```

Comment:
Shouldnt mediator.flush() be called from writeIndex?

Reply:
right... that would be better!
will update in the next rev!

Reply:
+1 on Jack comment.